### PR TITLE
fix(rust): fix csv reader with new line in header

### PR DESCRIPTION
**Pull request by uchiiii**
### Why
- To address https://github.com/pola-rs/polars/issues/8402

### CODE REVIEW

File: polars/tests/it/io/csv.rs [hunk 1]
```diff
@@ -340,6 +340,17 @@ fn test_new_line_escape() {
     let _df = CsvReader::new(file).has_header(true).finish().unwrap();
 }
 
+#[test]
+fn test_new_line_escape_on_header() {
+    let s = r#""length","header with
+ new line character", "width"
+ 5.1,3.5,1.4
+ "#;
+    println!("{}", "ok");
```
Comment (ritchie46) [inline polars/tests/it/io/csv.rs:349]: Can you remove the println?

File: polars/tests/it/io/csv.rs [hunk 2]
```diff
@@ -340,6 +340,17 @@ fn test_new_line_escape() {
...
+    let file: Cursor<&str> = Cursor::new(s);
+    let _df: DataFrame = CsvReader::new(file).has_header(true).finish().unwrap();
```
Comment (ritchie46) [inline polars/tests/it/io/csv.rs:351]: I think we should assert the output of the `DataFrame` to be correct for this test to have utility.

**Review by ritchie46 — COMMENTED** [review]
Thanks @uchiiii. I think the code looks good, but the test needs a bit more attention.

File: polars/tests/it/io/csv.rs (hunk 2 above)
Comment (uchiiii) [reply polars/tests/it/io/csv.rs]: I modified an assertion to the test. Review when you have time, please :)

**Comment by ritchie46** [comment]
Thanks @uchiiii!

**Pull request merged**